models/cfp.py

```diff
@@ -792,6 +803,55 @@ class InstallationProposal(Proposal):
     funds = db.Column(db.String, nullable=True)
 
 
+class LightningTalkProposal(Proposal):
+    __mapper_args__ = {"polymorphic_identity": "lightning"}
+    human_type = HUMAN_CFP_TYPES["lightning"]
+    slide_link = db.Column(db.String, nullable=True)
+    # This should probably use allowed_times or something but that has a very
+    # specific format and this is easier
+    session = db.Column(db.String, default="fri")
+
+    @classmethod
+    def get_remaining_lightning_slots(cls):
+        # Find which day's sessions still have spaces
+
+        slots = cls.get_total_lightning_talk_slots()
+
+        day_counts = {
+            day: count
+            for (day, count) in cls.query.with_entities(
+                cls.session,
+                func.count(cls.id),
+            )
+            .filter(cls.state != "withdrawn")
+            .group_by(cls.session)
+            .all()
+        }
+        return {
+            day: (
+                slots.get(day, (120 / LIGHTNING_TALK_LENGTH)) - day_counts.get(day, 0)
+            )
+            for day in LIGHTNING_TALK_SESSIONS.keys()
+        }
+
+    @classmethod
+    def get_total_lightning_talk_slots(cls):
+        sessions = TalkProposal.query.filter(
+            TalkProposal.title.startswith("Lightning Talk"),
```

Comment:
Feels like we should have a PlaceholderProposal or something, that inherits directly from TalkProposal. Can refactor before next time, though.

Reply:
yea, in theory next time we could add a further constraint to Slot Machine that forces all lightning talks for a single day to be together (with no time between) then they're basically 'just' a talk but will have to do some thinking... later